code/modules/urist/items/guns.dm

```diff
@@ -421,3 +421,255 @@ the sprite and make my own projectile -Glloyd*/
 	mag_type = MAGAZINE
 	multiple_sprites = 0
 
+/obj/item/weapon/gun/projectile/manualcycle/
+	var/bolt_open = 0
+
+/obj/item/weapon/gun/projectile/manualcycle/update_icon()
+	if(bolt_open)
+		icon_state = "308bolt_alt"
+	else
+		icon_state = "308bolt"
+
+/obj/item/weapon/gun/projectile/manualcycle/attack_self(mob/user as mob)
+	playsound(src.loc, 'sound/weapons/flipblade.ogg', 50, 1)
+	bolt_open = !bolt_open
+	if(bolt_open)
+		if(chambered)
+			user << "<span class='notice'>You work the bolt open, ejecting [chambered]!</span>"
+			chambered.loc = get_turf(src)
+			loaded -= chambered
+			chambered = null
+		else
+			user << "<span class='notice'>You work the bolt open.</span>"
+	else
+		user << "<span class='notice'>You work the bolt closed.</span>"
+		bolt_open = 0
+	add_fingerprint(user)
+	update_icon()
+
+/obj/item/weapon/gun/projectile/manualcycle/special_check(mob/user)
+	if(bolt_open)
+		user << "<span class='warning'>You can't fire [src] while the bolt is open!</span>"
+		return 0
+	return ..()
+
+/obj/item/weapon/gun/projectile/manualcycle/load_ammo(var/obj/item/A, mob/user)
+	if(!bolt_open)
+		return
+	..()
+
+/obj/item/weapon/gun/projectile/manualcycle/unload_ammo(mob/user, var/allow_dump=1)
+	if(!bolt_open)
+		return
+	..()
+
+/obj/item/weapon/gun/projectile/manualcycle/imprifle
+	name = "improvised rifle"
+	icon = 'icons/urist/items/guns.dmi'
+	desc = "A shoddy 7.62 improvised rifle."
+	wielded_item_state = "woodarifle-wielded"
+	icon_state = "308bolt"
+	item_state = "dshotgun" //placeholder
+	w_class = 5
+	requires_two_hands = 4
+	force = 10
+	slot_flags = SLOT_BACK
+	origin_tech = "combat=4;materials=1;syndicate=2"
```

Comment:
Definitely not Syndicate. You could drop combat down to 2, that would make more sense for something improvised.